.github/workflows/goreleaser.yml

```diff
@@ -7,6 +7,9 @@ on:
 jobs:
   goreleaser:
     runs-on: ubuntu-latest
+    # We want to run on external PRs, but not on our own internal PRs as they'll be run
+    # by the push to the branch.
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
```

Comment:
Shouldn't we also avoid running this for external PRs? I guess what's our full intent?